project-koku/koku
# [COST-5328] Fully distribute network unattributed costs

## Jira Ticket

[COST-5238](https://issues.redhat.com/browse/COST-5328)
[COST-3761](https://issues.redhat.com/browse/COST-3761)

## Description

Network unattributed costs are distributed in part by filtering for `cost_category_id IS NULL` in order to exclude Platform costs from Network unattributed. When putting network costs into their own project, the cost category ID is preserved. This results in some Network unattributed costs not being distributed to user projects.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load test customer data with a cost model set to distribute network costs
4. `curl "http://localhost:8000/api/cost-management/v1/reports/openshift/costs/?filter%5Bresolution%5D=monthly&group_by%5Bproject%5D=network"`
5. Verify that `data[0].projects[0].values[0].cost.total.value` + `data[0].projects[0].values[0].cost.network_unattributed_distributed.value` = 0
<details>
  <summary>JSON response snippet</summary>

```json
...
                            "cost": {
                                "raw": {
                                    "value": 31.152,
                                    "units": "USD"
                                },
                                "markup": {
                                    "value": 3.1152,
                                    "units": "USD"
                                },
                                "usage": {
                                    "value": 0.0,
                                    "units": "USD"
                                },
                                "platform_distributed": {
                                    "value": 0.0,
                                    "units": "USD"
                                },
                                "worker_unallocated_distributed": {
                                    "value": 0.0,
                                    "units": "USD"
                                },
                                "network_unattributed_distributed": {
                                    "value": -34.2672,
                                    "units": "USD"
                                },
                                "storage_unattributed_distributed": {
                                    "value": 0.0,
                                    "units": "USD"
                                },
                                "distributed": {
                                    "value": 0.0,
                                    "units": "USD"
                                },
                                "total": {
                                    "value": 34.2672,
                                    "units": "USD"
                                }
...
```
</details>

## Release Notes

```markdown
* [COST-5328](https://issues.redhat.com/browse/COST-5328) Fully distribute Network unattributed costs
```
